tools/mkfs.c

```diff
@@ -500,7 +500,9 @@ int main(int argc, char **argv)
     }
     if (!img_size)
         img_size = current_size;
-    img_size = pad(img_size, TFS_LOG_DEFAULT_EXTENSION_SIZE);
+    if ((img_size - offset) & (TFS_LOG_DEFAULT_EXTENSION_SIZE - 1))
```

Comment:
Maybe simpler to do "img_size = pad(img_size - offset, TFS_LOG_DEFAULT_EXTENSION_SIZE) + offset;" here?

Reply:
Your suggestion indeed makes the code simpler, thanks.